spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -71,12 +75,17 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		List<Binding<MessageChannel>> bindings = new ArrayList<>();
 		Binder<MessageChannel, ConsumerProperties, ?> binder =
 				(Binder<MessageChannel, ConsumerProperties, ?>) getBinderForChannel(inputChannelName);
-		Class<? extends ConsumerProperties> propertiesClass = resolveConsumerPropertiesType(binder);
 		ConsumerProperties consumerProperties =
-				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName, propertiesClass);
+				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
+		if (binder instanceof ExtendedPropertiesBinder) {
+			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
+			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
+			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
```

Comment:
Maybe IDEA is not as obnoxious about type safety, but I am seeing lots of warnings in eclipse - needs generics or at least `@SuppressWarning` - my OCD doesn't tolerate well the yellow squigglies.